.github/workflows/publish.yml

```diff
@@ -0,0 +1,117 @@
+        with:
+          aws-region: ${{ secrets.AWS_REGION }}
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+      - uses: actions/download-artifact@v3
+        with:
+          name: binaries
+      - run: |
+          for PACKAGE in *.deb *.rpm; do
+            curl -fs -F package=@$PACKAGE https://${{ secrets.GEMFURY_TOKEN }}@push.fury.io/infrahq/
+          done
+        if: ${{ inputs.ENVIRONMENT == 'production' }}
+      - run: aws s3 sync . s3://${{ secrets.RELEASES_BUCKET }}/infra/v${{ inputs.RELEASE_NAME }} --exclude '*' --include '*.zip' --include '*.deb' --include '*.rpm' --include 'infra-checksums.txt' --acl public-read
+      - uses: actions/download-artifact@v3
+        with:
+          name: helm
+      - run: aws s3 sync . s3://${{ secrets.HELM_BUCKET }} --exclude '*' --include 'index.yaml' --include '*.tgz' --acl public-read
+      - run: |
+          git config --global user.name infrahq-ci
+          git config --global user.email contact@infrahq.com
+      - uses: actions/checkout@v3
+        with:
+          repository: infrahq/homebrew-tap
+          path: homebrew-tap
+          token: ${{ secrets.RELEASE_PLEASE_TOKEN }}
+        if: ${{ inputs.ENVIRONMENT == 'production' }}
+      - run: |
+          sh update-tap.sh -b https://${{ inputs.RELEASES_BUCKET }}/infra $RELEASE_NAME
+          RECIPE=$(git status --porcelain | awk '{ print $2 }')
+          git add $RECIPE
+          git commit -m "Brew formula update for ${RECIPE%.*} version $RELEASE_NAME"
+          git push origin @:refs/heads/main
+        working-directory: homebrew-tap
+        if: ${{ inputs.ENVIRONMENT == 'production' }}
+      - uses: actions/checkout@v3
+        with:
+          repository: infrahq/scoop
+          path: scoop
+          token: ${{ secrets.RELEASE_PLEASE_TOKEN }}
+        if: ${{ inputs.ENVIRONMENT == 'production' }}
+      - run: |
+          sh update-scoop.sh -b https://${{ inputs.RELEASES_BUCKET }}/infra $RELEASE_NAME
+          PACKAGE=$(git status --porcelain | awk '{ print $2 }')
+          git add $PACKAGE
+          git commit -m "Scoop update for ${PACKAGE%.*} version $RELEASE_NAME"
+          git push origin @:refs/heads/main
+        working-directory: scoop
+        if: ${{ inputs.ENVIRONMENT == 'production' }}
+
+  publish-docker:
+    runs-on: ubuntu-latest
+    environment: ${{ inputs.ENVIRONMENT }}
+    steps:
+      - uses: actions/checkout@v3
+      - uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_PASSWORD }}
+      - id: metadata
+        uses: docker/metadata-action@v3
+        with:
+          images: infrahq/infra
+          flavor: latest=false
+          tags: |
+            type=raw,value=${{ inputs.RELEASE_NAME }}
+            type=raw,value=latest,enable={{is_default_branch}}
```

Comment:
This should, in theory, tag `latest` if the action is trigger on the main branch